### DISCUSSION TITLE
What is the nature of Prisma's dependency on async_hooks ?

### DISCUSSION THREAD

**Discussion by doronrk** [prisma/prisma#14823] [Q&A] [answered]
What is the nature of Prisma's dependency on async_hooks ? Follow up question - can it be circumvented?
I'm assuming the answer to the second question is "no". The reason I'm asking is because I'm attempting to migrate my app off of Meteor by incrementally adopting Prisma to replace Meteor collections. Meteor, unfortunately, uses fibers and can lead to memory leaks when used with async_hooks.  We could move forward with one massive commit that introduces Prisma and removes Meteor, but incremental adoption would be preferable. Incremental adoption poses the problem of adding async_hooks into a Meteor environment, which leads to my questions above.
Thanks!

**Comment by austincrim** [comment, marked as answer]
Hey @doronrk 👋,

Prisma uses `async_hooks` when executing Prisma middleware. It might be possible to sidestep the dependency using something like [`patch-package`](https://www.npmjs.com/package/patch-package), but that would add some maintenance burden

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
